master/internal/user/postgres_users_intg_test.go

```diff
@@ -0,0 +1,358 @@
+			require.NoError(t, err)
+			require.Equal(t, uid, resGroup.OwnerID)
+			require.Equal(t, fmt.Sprintf("%d%s", uid, PersonalGroupPostfix), resGroup.Name)
+
+			// require entry in the group membership table (addUser).
+			err = db.Bun().NewSelect().Table("user_group_membership").
+				Where("user_id = ?", uid).Scan(context.TODO(), &resGroupMem)
+			require.NoError(t, err)
+			require.Equal(t, uid, resGroupMem.UserID)
+			require.Equal(t, resGroup.ID, resGroupMem.GroupID)
+
+			// require entry in the AUG membership table (addAgentUserGroup).
+			// But also check that the AUG isn't nil/empty.
+			if tt.ug != (model.AgentUserGroup{}) {
+				err = db.Bun().NewSelect().Table("agent_user_groups").
+					Where("user_id = ?", uid).Scan(context.TODO(), &resAUG)
+				require.NoError(t, err)
+				require.Equal(t, uid, resAUG.UserID)
+				require.Equal(t, tt.ug.User, resAUG.User)
+				require.Equal(t, tt.ug.UID, resAUG.UID)
+			}
+		})
+	}
+}
+
+func TestUserUpdate(t *testing.T) {
+	cases := []struct {
+		name        string
+		ug          model.AgentUserGroup
+		updatedUser model.User
+	}{
+		{
+			"simple-case",
+			model.AgentUserGroup{},
+			model.User{Username: uuid.NewString()},
+		},
+		{
+			"aug-defined-1",
+			model.AgentUserGroup{User: uuid.NewString(), UID: 1},
+			model.User{Username: uuid.NewString(), Admin: false, Active: false, Remote: false},
+		},
+		{
+			"aug-defined-2",
+			model.AgentUserGroup{User: uuid.NewString(), UID: 123},
+			model.User{Username: uuid.NewString(), Admin: true, Active: true, Remote: true},
+		},
+	}
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			user, err := addTestUser(nil)
+			require.NoError(t, err)
+
+			// Test Update.
+			tt.updatedUser.ID = user.ID
+			err = Update(context.TODO(), &tt.updatedUser, []string{"admin", "active", "remote"}, &tt.ug)
+			require.NoError(t, err)
+
+			// Now require that the fields are indeed updated.
+			var updatedUser model.User
+			err = db.Bun().NewSelect().Table("users").
+				Where("id = ?", tt.updatedUser.ID).Scan(context.TODO(), &updatedUser)
+			require.NoError(t, err)
+			require.Equal(t, tt.updatedUser.Admin, updatedUser.Admin)
+			require.Equal(t, tt.updatedUser.Active, updatedUser.Active)
+			require.Equal(t, tt.updatedUser.Remote, updatedUser.Remote)
+		})
+	}
+}
+
+func TestUserStartSession(t *testing.T) {
+	user, err := addTestUser(nil)
+	require.NoError(t, err)
+
+	// Add a session.
+	var session model.UserSession
+	token, err := StartSession(context.TODO(), user)
+	require.NoError(t, err)
+	require.NotNil(t, token)
+
+	err = db.Bun().NewSelect().Table("user_sessions").
+		Where("user_id = ?", user.ID).Scan(context.TODO(), &session)
+	require.NoError(t, err)
+}
+
+func TestDeleteSessionByToken(t *testing.T) {
+	userID, _, token, err := addTestSession()
+	require.NoError(t, err)
+
+	err = DeleteSessionByToken(context.TODO(), token)
+	require.NoError(t, err)
+	require.Error(t, db.Bun().NewSelect().Table("user_sessions").
```

Comment:
could use ```require.ErrorAs``` here for whatever bun uses for the not found error

The only concern is that our tests are passing but they are somehow broken and we didn't know